src/coreclr/debug/di/shimstackwalk.cpp

```diff
@@ -988,6 +989,17 @@ CorDebugInternalFrameType ShimStackWalk::GetInternalFrameType(ICorDebugInternalF
 
 void ShimStackWalk::AppendFrame(ICorDebugFrame * pFrame, StackWalkInfo * pStackWalkInfo)
 {
+    if (pStackWalkInfo->m_fHasException && pStackWalkInfo->m_cFrame == 0)
```

Comment:
Lets add a comment describing why this is necessary

Reply:
Added the comment, please let me know if it's clear enough?